### DISCUSSION TITLE
Way Users Not Automatically Logged Out After Being Banned

### DISCUSSION THREAD

**Discussion by arashsaffari** [codeigniter4/shield#1200] [Q&A] [answered]
### PHP Version

8.3

### CodeIgniter4 Version

4.5.5

### Shield Version

1.1

### Which operating systems have you tested for this bug?

macOS

### Which server did you use?

apache

### Database

MariaDB 10.2

### Did you customize Shield?

No

### What happened?

When I ban a user, if the user is still logged in, they are not automatically logged out. This means they remain logged in until they log out themselves or the session expires.

### Steps to Reproduce

-

### Expected Output

-

**Comment by datamweb** [comment, marked as answer]
If you're using session-based authentication, you need to ensure that the session filter is correctly set up to automatically log out banned users.

https://shield.codeigniter.com/quick_start_guide/using_session_auth/#protecting-pages

**Comment by kenjis** [comment]
Banning feature just changes the user status. It does not log the user out.
See https://shield.codeigniter.com/user_management/banning_users/

**Comment by arashsaffari** [comment]
> If you're using session-based authentication, you need to ensure that the session filter is correctly set up to automatically log out banned users.
> 
> https://shield.codeigniter.com/quick_start_guide/using_session_auth/#protecting-pages


That was the problem. Now that I have set the session filter, it works again. Thank you!
